appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockComponentsUtil.java

```diff
@@ -259,7 +262,7 @@ private static String getHexString(String color, int digits) {
     int len = color.length();
     if (len < digits) {
       do {
-        color = '0' + color;
+        color = color + '0';
```

Comment:
This change has a lot of implications for calling functions. I would recommend not making this change here and updating `setWidgetBackgroundColor` to pull the alpha from the first two hex values rather than the last two values, using the `rgba` color constructor rather than the `#` based approach.

Reply:
You are right, the changes in `getHexString` were not unnecessary. They were meant to make the `getHexString` function return in ARGB format, but we would have to convert the functions which call it to also be written in the same format.
If we simply remove the changes in the `getHexString` method from the commit, the bug remains fixed and the `setWidgetBackgroundColor` works as expected.
So should I just remove the `getHexString` changes?

Reply:
@AnujSinghal12 If that's sufficient, then I would recommend it. However, it would be good if we understand the mechanism as to why it works.

Reply:
When we pass the "color" string and "8" as the parameters into `getHexSting`, it simply removes the "&H" part from the start without going inside the loop in the function. Therefore changing the functionality of the loop inside `getHexString` does not affect the `setWidgetBackgroundColor` in my commit.
I'll add the commit that removes the changes in `getHexString`.